db/sqlinterfaces.c

```diff
@@ -6344,6 +6344,51 @@ static inline int tdef_to_tranlevel(int tdef)
     }
 }
 
+void cleanup_clnt(struct sqlclntstate *clnt)
+{
+    if (clnt->saved_errstr) {
+        free(clnt->saved_errstr);
+        clnt->saved_errstr = NULL;
+    }    
+
+    if (clnt->context) {
+        for (int i = 0; i < clnt->ncontext; i++) {
+            free(clnt->context[i]);
+        }
+        free(clnt->context);
+        clnt->context = NULL;
+    }
+
+
+    if (clnt->selectv_arr) {
+        currangearr_free(clnt->selectv_arr);
+        clnt->selectv_arr = NULL;
+    }
+
+    if (clnt->arr) {
+        currangearr_free(clnt->arr);
+        clnt->arr = NULL;
+    }
+
+    if (clnt->spversion.version_str) {
```

Comment:
Not sure we should do this.. If next run matches sp + version, then there is no need to fetch sp code again.

Reply:
reset_clnt() does the following:
```
clnt->spversion.version_num = 0;
free(clnt->spversion.version_str);
clnt->spversion.version_str = NULL;
```
So i think that right before clnt destruction this should be not an issue. Doing so in reset_clnt() may need another look?

Reply:
OK to do if clnt is getting destroyed.

Reply:
Sorry! This is OK to do (even during reset) - the version check is done against clnt->sp.